src/stores/FileBrowserStore.ts

```diff
@@ -369,6 +375,36 @@ export class FileBrowserStore {
         this.selectedFiles = selection;
     };
 
+    @action showLoadingDialog = () => {
+        this.isLoadingDialogOpen = true;
+    }
+
+    @action updateLoadingState = (progress: number, checkedCount: number, totalCount: number) => {
+        this.loadingProgress = progress;
+        this.loadingCheckedCount = checkedCount;
+        this.loadingTotalCount = totalCount;
+    };
+
+    @action resetLoadingStates = () => {
+        this.loadingList = false;
+        this.isLoadingDialogOpen = false;
+        this.updateLoadingState(0, 0, 0);
+    };
+
+    @action cancelRequestingFile = () => {
```

Comment:
can't we just combine this into one function? `this.browserMode` can be used to determine the `FileListType`